frontend/packages/ceph-storage-plugin/src/types.ts

```diff
@@ -153,6 +153,39 @@ export type KMSConfig = {
   hasHandled: boolean;
 };
 
+export type IbmKmsConfig = {
+  name: {
+    value: string;
+    valid: boolean;
+  };
+  instanceId: {
+    value: string;
+    valid: boolean;
+  };
+  apiKey: {
+    value: string;
+    valid: boolean;
+  };
+  rootKey: {
+    value: string;
+    valid: boolean;
+  };
+  baseUrl: string;
+  tokenUrl: string;
+  hasHandled: boolean;
+};
+
+export enum ProviderNames {
+  vault = 'vault',
+  ibmKms = 'ibmKms',
+}
+
+export type KMSConfig = {
+  vault: VaultConfig;
+  ibmKms: IbmKmsConfig;
+  kmsProvider: ProviderNames;
```

Comment:
can we move "hasHandled: boolean" under KMSConfig, let keep all common keys/values under KMSConfig instead of duplicating on each config.

Also instead of maintaining two different configs under two different keys ```vault`` and ```ibmKms```

what about
```
type IbmConfig = {
  name: {
    value: string;
    valid: boolean;
  };
  token?: {
    value: string;
    valid: boolean;
  };
  address: {
    value: string;
    valid: boolean;
  };
  port: {
    value: string;
    valid: boolean;
  };
  backend: string;
  clientKeyFile: string;
  providerNamespace: string;
};

type VaultConfig = {
  name: {
    value: string;
    valid: boolean;
  };
  token?: {
    value: string;
    valid: boolean;
  };
  address: {
    value: string;
    valid: boolean;
  };
  port: {
    value: string;
    valid: boolean;
  };
  backend: string;
  clientKeyFile: string;
  providerNamespace: string;
};

const VaultEmptyState: VaultConfig = Object.seal({
  name: {
    value: 'Vault',
    valid: true,
  },
  token: {
    value: '',
    valid: true,
  },
  address: {
    value: '',
    valid: true,
  },
  port: {
    value: '',
    valid: true,
  },
  backend: '',
  providerNamespace: '',
  clientKeyFile: '',
});

const IbmEmptyState: VaultConfig = Object.seal({
  name: {
    value: 'IBM',
    valid: true,
  },
  token: {
    value: '',
    valid: true,
  },
  address: {
    value: '',
    valid: true,
  },
  port: {
    value: '',
    valid: true,
  },
  backend: '',
  providerNamespace: '',
  clientKeyFile: '',
});


type KmsServiceMapping = {
  vault: VaultConfig;
  ibm: IbmConfig;

};

type KmsConfig = {
  kmsService: KmsServiceMapping;
  hasEnabled: boolean;
  kmsProvider: string;
};

const kmsConfig: KmsConfig = {
  kmsService: {
    vault: VaultEmptyState,
    ibm: IbmEmptyState
  },
  hasEnabled: true,
  kmsProvider: "vault"
};
```

Reply:
So I can access using the same key 

kmsConfig.kmsService[kmsConfig.kmsProvider]
kmsConfig.hasEnabled

so not much change is required when new kms service is introduced in future

Reply:
above one is just a proposal, i am not used exact keys which we have  in our console, just a idea

Reply:
outside of your KMS service component, we can use 
kmsConfig.kmsService[kmsConfig.kmsProvider]

Inside a specific KMS component we can use like
kmsConfig.kmsService[VAULT] (or) kmsConfig.kmsService[IBM]

Reply:
ack... will make the change... should not be a lot of changes...

Reply:
No need to move handled outside, let it be provider-specific. now i got it why you are keeping inside provider config

Reply:
I checked other PR, it LGTM, no need to do this change